library/alloc/src/slice.rs

```diff
@@ -782,6 +782,38 @@ impl<T, A: Allocator> BorrowMut<[T]> for Vec<T, A> {
     }
 }
 
+// Specializable trait for implementing ToOwned::clone_into. This is
+// public in the crate and has the Allocator parameter so that
+// vec::clone_from use it too.
+#[cfg(not(no_global_oom_handling))]
+pub(crate) trait SpecCloneIntoVec<T, A: Allocator> {
```

Comment:
...so that this doesn't need to be exported?  And can thus instead be
```suggestion
trait SpecCloneIntoVec<T, A: Allocator> {
```